Sources/TuistCore/MetadataProviders/XCFrameworkMetadataProvider.swift

```diff
@@ -128,7 +128,7 @@ public final class XCFrameworkMetadataProvider: PrecompiledMetadataProvider, XCF
         case "framework":
             binaryPath = AbsolutePath(library.identifier, relativeTo: xcframeworkPath)
                 .appending(RelativePath(library.path.pathString))
-                .appending(component: binaryName)
+                .appending(component: library.path.pathString.replacingOccurrences(of: ".framework", with: ""))
```

Comment:
This may not achieve the desired results as `pathString` is a path rather than a file name. 

I believe `.basenameWithoutExt` may be more appropriate.

Reply:
You were right. Thanks!